src/stream_servers/xiu.rs

```diff
@@ -0,0 +1,252 @@
+
+        let client = &self.client;
+        let mut request = client
+            .post(&self.stats_url)
+            .header("Content-Type", "application/json");
+
+        request = request.json(&body);
+
+        let res = match request.send().await {
+            Ok(res) => res,
+            Err(_) => {
+                error!("Xiu API ({}) is unreachable", self.stats_url);
+                return None;
+            }
+        };
+
+        if res.status() != reqwest::StatusCode::OK {
+            error!("Error accessing Xiu API ({})", self.stats_url);
+            return None;
+        }
+
+        let text = res.text().await.ok()?;
+        let data: XiuResponse = serde_json::from_str(&text).ok()?;
+
+        if data.error_code != 0 {
+            error!("Error accessing Xiu API ({}) {}", self.stats_url, data.desp);
+            return None;
+        }
+
+        if data.data.is_empty() {
+            error!("No data returned from Xiu API ({})", self.stats_url);
+            return None;
+        }
+
+        let publisher = serde_json::to_value(&data.data[0].publisher).ok()?;
+
+        let stream: XiuPublisher = match serde_json::from_value(publisher.to_owned()) {
+            Ok(stats) => stats,
+            Err(error) => {
+                trace!("{:?}", &data);
+                error!("Error parsing stats ({}) {}", self.stats_url, error);
+                return None;
+            }
+        };
+
+        trace!("{:#?}", stream);
+        Some(stream)
+    }
+}
+
+#[derive(Deserialize, Debug)]
+struct XiuResponse {
+    error_code: i32,
+    desp: String,
+    data: Vec<XiuStreamInfo>,
+}
+
+#[async_trait]
+#[typetag::serde]
+impl SwitchLogic for Xiu {
+    /// Which scene to switch to
+    async fn switch(&self, triggers: &Triggers) -> SwitchType {
+        let stats = match self.get_stats().await {
+            Some(b) => b,
+            None => return SwitchType::Offline,
+        };
+
+        if let Some(offline) = triggers.offline {
+            if stats.recv_bitrate > 0 && stats.recv_bitrate <= offline.into() {
+                return SwitchType::Offline;
+            }
+        }
+
+        if stats.recv_bitrate == 0 {
+            return SwitchType::Previous;
+        }
+
+        if let Some(low) = triggers.low {
+            if stats.recv_bitrate <= low.into() {
+                return SwitchType::Low;
+            }
+        }
+
+        return SwitchType::Normal;
+    }
+}
+
+#[async_trait]
+#[typetag::serde]
+impl StreamServersCommands for Xiu {
+    async fn bitrate(&self) -> super::Bitrate {
+        let stats = match self.get_stats().await {
+            Some(stats) => stats,
+            None => return super::Bitrate { message: None },
+        };
+
+        if !stats.video.is_some() {
+            return super::Bitrate { message: None };
+        }
+
+        let video_bitrate = stats.video.as_ref().unwrap().bitrate;
+        super::Bitrate {
+            message: Some(format!("{} Kbps", video_bitrate)),
```

Comment:
Kbps is not included in other stream servers on the bitrate command.

```suggestion
            message: Some(format!("{}", video_bitrate)),
```